workspaces/cli-server/src/routers/capture-router.ts

```diff
@@ -152,62 +147,29 @@ export function makeRouter(dependencies: ICaptureRouterDependencies) {
 
   router.get('/diffs/:diffId/diffs', async (req, res) => {
     const { captureId, diffId } = req.params;
-    const diffOutputPaths = getDiffOutputPaths({
-      captureBaseDirectory: req.optic.paths.capturesPath,
-      captureId,
-      diffId,
-    });
 
-    try {
-      if (fs.existsSync(diffOutputPaths.diffsStream)) {
-        const diffsJson = chain([
-          fs.createReadStream(diffOutputPaths.diffsStream),
-          jsonlParser(),
-          (data) => [data.value],
-          jsonDisassembler(),
-          jsonStringer({ makeArray: true }),
-        ]);
-
-        diffsJson.pipe(res).type('application/json');
-      } else {
-        //@TODO: streamify
-        await lockfile.lock(diffOutputPaths.diffs, {
-          retries: { retries: 10 },
-        });
-        const contents = await fs.readJson(diffOutputPaths.diffs);
-        await lockfile.unlock(diffOutputPaths.diffs);
-        res.json(contents);
-      }
-    } catch (e) {
-      res.status(404).json({
-        message: e.message,
-      });
+    const diffMetadata = diffs.get(diffId);
+    if (!diffMetadata) {
+      return res.json(404);
     }
+    const diffQueries = diffMetadata.manager.queries();
+
+    let diffsStream = diffQueries.diffs();
+    diffsStream.pipe(toJSONArray()).pipe(res).type('application/json');
```

Comment:
is there a streamy way to wrap it in an object to preserve the original shape?

Reply:
There totally should be and is, by prepending and appending [`stream-json` type tokens ](https://github.com/uhop/stream-json/wiki/Parser#stream-of-tokens) that describe this, but I bailed on that after 2 attempts of getting that to work right. That was while dealing some other issues too, so perhaps it's worth a revisit.

Reply:
Found a way to implement this with and understandable API in https://github.com/opticdev/optic/pull/371/commits/60d6e55f883daca812f910fa688358c7d97438bb.